src/main/java/com/thebois/models/beings/roles/RoleFactory.java

```diff
@@ -75,37 +91,48 @@ public static AbstractRole guard() {
     /**
      * Creates a new miner role.
      *
-     * @return A new miner role
+     * @return A new miner role.
      */
     public static AbstractRole miner() {
-        return new MinerRole();
+        assertDependenciesNotNull();
+        return new MinerRole(resourceFinder, world);
     }
 
     /**
      * Creates a new fisher role.
      *
-     * @return A new fisher role
+     * @return A new fisher role.
      */
     public static AbstractRole fisher() {
-        return new FisherRole();
+        assertDependenciesNotNull();
+        return new FisherRole(resourceFinder, world);
     }
 
     /**
      * Creates a new builder role.
      *
-     * @return A new builder role
+     * @return A new builder role.
      */
     public static AbstractRole builder() {
         return new BuilderRole();
     }
 
     /**
-     * Creates a new idle role.
+     * Creates a new idle role with random movement.
      *
-     * @return A new idle role
+     * @return A new randomly moving idle role.
      */
     public static AbstractRole idle() {
-        return new IdleRole();
+        assertDependenciesNotNull();
+        return new IdleRole(world);
+    }
+
+    private static void assertDependenciesNotNull() {
+        Objects.requireNonNull(world,
+                               "World can not be null. Call RoleFactory.setWorld to set it.");
+        Objects.requireNonNull(
+            resourceFinder,
+            "Resource Finder can not be null. Call RoleFactory.setResourceFinder to set it.");
     }
```

Comment:
Question
This method is being called only to make sure that every role has a world and finder, correct? 
Mostly asking because I've never seen a method like this before.